x/gamm/keeper/pool.go

```diff
@@ -198,8 +198,14 @@ func (k Keeper) DeletePool(ctx sdk.Context, poolId uint64) error {
 // 	return nil
 // }
 
+// GetPoolDenom retrieves the pool based on PoolId and
+// returns the coin denoms that it holds.
 func (k Keeper) GetPoolDenoms(ctx sdk.Context, poolId uint64) ([]string, error) {
 	pool, err := k.GetPoolAndPoke(ctx, poolId)
+	if err != nil {
+		return nil, err
```

Comment:
let's please also add a case where poolId does not exist so that we cover this branch

Reply:
added here: [`c59f147`](https://github.com/osmosis-labs/osmosis/pull/2240/commits/c59f147bb825ef80b00dda548d7261e3a39cb2be)